modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -11,27 +11,21 @@ def get_claims(veis_token, btsss_token, params = {})
       {
         data: claims.map do |sc|
           sc['claimStatus'] = sc['claimStatus'].underscore.titleize
+          sc['id'] = Digest::SHA256.hexdigest(sc['id'])
           sc
         end
       }
     end
 
     def get_claim_by_id(veis_token, btsss_token, claim_id)
-      # ensure claim ID is the right format, allowing any version
-      uuid_all_version_format = /^[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[89ABCD][0-9A-F]{3}-[0-9A-F]{12}$/i
-
-      unless uuid_all_version_format.match?(claim_id)
-        raise ArgumentError, message: "Expected claim id to be a valid UUID, got #{claim_id}."
-      end
-
       claims_response = client.get_claims(veis_token, btsss_token)
 
       claims = claims_response.body['data']
 
-      claim = claims.find { |c| c['id'] == claim_id }
-
+      claim = claims.find { |c| Digest::SHA256.hexdigest(c['id']) == claim_id }
```

Comment:
I didn't abstract the `Digest::SHA256.hexdigest` calls into their own module/class because I'm not sure if it will be as widely used as the [VAOS reference implementation](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/vaos/app/helpers/vaos/anonymizers.rb). If we have any error logging added as part of https://github.com/department-of-veterans-affairs/va.gov-team/issues/93792, I think we will have to go the [`PersonalInformationLog`](https://depo-platform-documentation.scrollhelp.site/developer-docs/personal-identifiable-information-pii-guidelines#PersonalIdentifiableInformation(PII)guidelines-PIIwiththePersonalInformationLog) route.